src/array.cr

```diff
@@ -947,13 +922,9 @@ class Array(T)
     end
   end
 
-  # Replaces every element in `self` with the given *value*. Returns `self`.
-  #
-  # ```
-  # a = [1, 2, 3]
-  # a.fill(9) # => [9, 9, 9]
-  # ```
+  # :nodoc:
```

Comment:
This overrides a public method. I don't think it should be hidden as nodoc.
Ut should be `:inherit:` like `#rotate!` (plus an optional comment about the reason for the override).

Reply:
This comes from the overrides of `Indexable#index`, I think it is fine. If not there should be a separate issue about it.

Reply:
In any case, this PR should not add different approaches for overriding methods.

Overrides of public API methods are also public API and thus should be documented.